intern/service/dvid/versioning.py

```diff
@@ -0,0 +1,190 @@
+# Copyright 2019 The Johns Hopkins University Applied Physics Laboratory
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from intern.service.dvid import DVIDService
+import requests
+import numpy as np
+import json
+import ast
+
+
+class VersioningService(DVIDService):
+    """ VersioningService for DVID service.
+    """
+
+    def __init__(self, base_url):
+        """ Constructor.
+
+        Args:
+            base_url (str): Base url (host) of project service.
+
+        Raises:
+            (KeyError): if given invalid version.
+        """
+        DVIDService.__init__(self)
+        self.base_url = base_url
+
+    def merge(self, UUID, parents, mergeType, note):
+        """ Creates a conflict-free merge of a set of committed parent UUIDs into a child.  Note
+            the merge will not necessarily create an error immediately
+
+        Args:
+            mergeType (str) = "conflict-free"
```

Comment:
Is this like a commit message? Assume this is something a dvid user would know what's valid to provide?

Reply:
right